### ISSUE TITLE
Add ability to get private key for a address

### ISSUE THREAD

**Issue by thesimplekid** [bitcoindevkit/bdk#835]
**Describe the enhancement**  
Add a method that reveals the private key for a address in a wallet. 

**Use case**  
Used to prove ownership of a UTXO, this is necessary for a Joinmarket like [Podle](https://reyify.com/blog/poodle) verification, to prove that your counter party owns the UTXO they claim to.  

**Additional context**
Would mirror functionality to the bitcoincore rpc dump private key function 
https://docs.rs/bitcoincore-rpc/0.16.0/bitcoincore_rpc/trait.RpcApi.html#method.dump_private_key

**Comment by danielabrozzoni** [comment]
Hey there! I think it would be an useful method, even tho it should be available only for single sig wallets. At the moment we're working on the BDK 1.0 release, but after that, we can work on that :)

**Comment by thesimplekid** [comment]
Great, thank you

**Comment by notmandatory** [comment]
@thesimplekid is this something you still need?

**Comment by thesimplekid** [comment]
> @thesimplekid is this something you still need?

No, was an old project I was working on that I'm unlikely to pick back up.
